packages/ERTP/src/paymentLedger.js

```diff
@@ -507,13 +456,21 @@ export const preparePaymentLedger = (
       // @ts-expect-error checked cast
       newAmount = coerce(newAmount);
       mustMatch(newAmount, amountShape, 'minted amount');
-      const payment = makePayment();
-      initPayment(payment, newAmount, undefined);
+      // `rawPayment` is not associated with any recovery set, and
+      // so must not escape.
+      const rawPayment = makePayment();
+      initPayment(rawPayment, newAmount, undefined);
+
+      const mintRecoveryPurseBefore = mintRecoveryPurse.getCurrentAmount();
+      mintRecoveryPurse.deposit(rawPayment, newAmount);
+      const payment = mintRecoveryPurse.withdraw(newAmount);
+      const mintRecoveryPurseAfter = mintRecoveryPurse.getCurrentAmount();
+      assert(keyEQ(mintRecoveryPurseBefore, mintRecoveryPurseAfter));
```

Comment:
What does this protect us from?

Reply:
Only that after putting the payment in and then withdrawing a payment of the same amount, the purse's balance is unchanged. Just super extra paranoia on my part.

Resolving.